### PR TITLE
feat(table): initial commit

### PR DESCRIPTION
**Pull request by manoncarbonnel**
Added a new table component with its tests and accessibility recommendations guidelines in its documentation.

### CODE REVIEW

**Review by astagnol — CHANGES_REQUESTED** [review]
- please create storybook files for the component (it would have been created by using the generator)
- some comments refer to the Chip instead of Table
- as I understand why it could be nice to have recommendations in Usage Guidelines, this comes from the Zeroheight and I'm not sure we should duplicate it. We need to discuss a bit about this

File: packages/components/src/table/src/components/osds-table/core/controller.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import type { ODS_TABLE_SIZE } from '../constants/table-size';
+import type { ODS_TABLE_VARIANT } from '../constants/table-variant';
+import type { OsdsTable } from '../osds-table';
+import { OdsLogger } from '@ovhcloud/ods-common-core';
+import { ODS_TABLE_SIZES } from '../constants/table-size';
+import { ODS_TABLE_VARIANTS } from '../constants/table-variant';
+
+/**
+ * common controller logic for chip component used by the different implementations.
```
Comment (aesteves60) [inline packages/components/src/table/src/components/osds-table/core/controller.ts:9]: Remove comment

File: packages/components/src/table/src/components/osds-table/core/controller.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    this.component = component;
+  }
+
+  /**
+   * Warn user if the size attribute is not set
+   * @param size - size of the table
+   */
+  validateSize(size?: string): void {
+    if (!size || size.length === 0) {
+      this.logger.warn('The size attribute must be set');
```
Comment (aesteves60) [inline packages/components/src/table/src/components/osds-table/core/controller.ts:26]: Use console.warn and in the test spy it (everywhere)

File: packages/components/src/table/src/components/osds-table/osds-table.e2e.screenshot.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  describe('screenshots', () => {
+    beforeEach(() => {
+      variations = [];
+    });
+
+    it('should take screenshots of all attributes variations', async() => {
+      variations.push(`
+        <osds-table size='sm''></osds-table>
```
Comment (aesteves60) [inline packages/components/src/table/src/components/osds-table/osds-table.e2e.screenshot.ts:24]:
```suggestion
        <osds-table size='sm'></osds-table>
```

File: packages/components/src/table/src/components/osds-table/osds-table.e2e.screenshot.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      `);
+      variations.push(`
+        <osds-table size='md''></osds-table>
```
Comment (aesteves60) [inline packages/components/src/table/src/components/osds-table/osds-table.e2e.screenshot.ts:27]:
```suggestion
        <osds-table size='md'></osds-table>
```

File: packages/components/src/table/src/components/osds-table/osds-table.e2e.screenshot.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      `);
+      variations.push(`
+        <osds-table size='lg''></osds-table>
```
Comment (aesteves60) [inline packages/components/src/table/src/components/osds-table/osds-table.e2e.screenshot.ts:30]:
```suggestion
        <osds-table size='lg'></osds-table>
```

File: packages/components/src/table/src/components/osds-table/osds-table.e2e.screenshot.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      `);
+      variations.push(`
+        <osds-table size='striped''></osds-table>
```
Comment (aesteves60) [inline packages/components/src/table/src/components/osds-table/osds-table.e2e.screenshot.ts:33]:
```suggestion
        <osds-table size='striped'></osds-table>
```

**Review by aesteves60 — CHANGES_REQUESTED** [review]

File: packages/components/src/table/documentation/usage-guidelines/usage.mdx [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+    </tr>
+    </tbody>
+  </table>
+</osds-table>
+```
+
+### Size
+<SizeProperty />
+
+### Width
```
Comment (astagnol) [inline packages/components/src/table/documentation/usage-guidelines/usage.mdx:85]: Width -> Variant

**Review by astagnol — CHANGES_REQUESTED** [review]
- On Demo, we should be able to switch between default and striped Table

**Review by aesteves60 — APPROVED** [review]

**Review by astagnol — CHANGES_REQUESTED** [review]
Columns width acts a bit differently from native table. We can discuss about this

**Review by astagnol — APPROVED** [review]

**Review by dpellier — APPROVED** [review]

**Pull request merged**